From 247c662d49bd795c5450e00fe9325d69579920f0 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Thu, 2 May 2013 18:31:37 +0300
Subject: [PORT FROM MAIN] ril: Add radio off reason to RIL_REQUEST_RADIO_POWER

BZ: 102058

With this patch, radio off reason is added as
second parameter to the RIL_REQUEST_RADIO_POWER
request.

Based on the raddio off reason(INIT, SHUTDOWN, AIRPLANE
MODE), vendor ril can take necessary actions.

Severity: Medium

Change-Id: I4db703930b9d727dd9ece85c6a5002cdbabf5e1c
Orig-Change-Id: I2dab04a1302b288861285e3eb9ca59cf3948631b
Category: device enablement
Domain: Telephony-Modem-Common
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
---
 include/telephony/ril.h |    7 +++++++
 1 files changed, 7 insertions(+), 0 deletions(-)

diff --git a/include/telephony/ril.h b/include/telephony/ril.h
index 0848c19..4e5eaf8 100644
--- a/include/telephony/ril.h
+++ b/include/telephony/ril.h
@@ -94,6 +94,12 @@ typedef enum {
 } RIL_RadioState;
 
 typedef enum {
+    RIL_RADIO_OFF_REASON_NONE = 0,
+    RIL_RADIO_OFF_REASON_SHUTDOWN = 1,
+    RIL_RADIO_OFF_REASON_AIRPLANE_MODE = 2
+} RIL_Radio_Off_Reason;
+
+typedef enum {
     RADIO_TECH_UNKNOWN = 0,
     RADIO_TECH_GPRS = 1,
     RADIO_TECH_EDGE = 2,
@@ -1605,6 +1611,7 @@ typedef struct {
  * "data" is int *
  * ((int *)data)[0] is > 0 for "Radio On"
  * ((int *)data)[0] is == 0 for "Radio Off"
+ * ((int *)data)[1] if data[0] is 0, data[1] is a RIL_Radio_Off_Reason.
  *
  * "response" is NULL
  *
-- 
1.7.4.1

